qt/widgets/common/src/FitPropertyBrowser.cpp

```diff
@@ -1431,13 +1431,28 @@ void FitPropertyBrowser::stringChanged(QtProperty *prop) {
 
     QString parName = h->functionPrefix() + "." + parProp->propertyName();
 
-    QString str = m_stringManager->value(prop);
+    // Get the tie expression stored in the function in case the new expression is invalid
+    // and the GUI needs to be reset
+    const size_t parIndex = compositeFunction()->parameterIndex(parName.toStdString());
+    const auto oldTie = compositeFunction()->getTie(parIndex);
+    const std::string oldTieStr = oldTie->asString();
+    QString oldExp = QString::fromStdString(oldTieStr.substr(oldTieStr.find("=") + 1));
+
+    QString exp = m_stringManager->value(prop);
```

Comment:
I think the last two, you've removed the variable name. Just so I understand, what is the benefit to using `auto` here instead of specifying the type?

Reply:
Oh yes, thanks for catching that. 

`auto` is useful for removing the amount of re-factoring you have to do if there's a change a class name or return type or something. As long as the API for the class stays the same then if Qt decided that `fromStdString()` now returns a `QStringNew`, you wouldn't have to update your code as auto would work that out for you.

Reply:
Okay makes sense thank you